### DISCUSSION TITLE
download failed

### DISCUSSION THREAD

**Discussion by chenhoe95** [aclap-dev/video-downloadhelper#645] [Help and Q&A]
### Your question

{
  "version": "9.0.1.13",
  "target": "mozilla",
  "channel": "stable",
  "page_url": "https://taiav.com/cn/movie/5fec0047dfe76618649b16c7",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [tls @ 00000235032fa500] IO error: Error number -10054 occurred\r\n    Last message repeated 2 times\r\n[aac_adtstoasc @ 0000023502cd5d00] Error parsing ADTS frame header!\r\n[mp4 @ 0000023502ccf280] Error applying bitstream filters to an output packet for stream #1: Invalid data found when processing input\r\n[aost#0:1/copy @ 0000023502cfe1c0] Error submitting a packet to the muxer: Invalid data found when processing input\r\n[out#0/mp4 @ 0000023502d02400] Error muxing a packet\r\n",
    "id": "coapp_failure_9db6f33b-5f7d-4123-ad1f-32aaccea4ad7",
    "downloadable_id": "downloadable_6671026216610773"
  }
}

### Addon & CoApp version and details

version: 9.0.1.13
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "download_directory": "C:\\Users\\Frankie\\Downloads\\Video",
    "concurrent_downloads_max": 10,
    "show_success_notification": false,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": false,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_wide_ui": true,
    "open_count_store": 144
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

I am running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.22a1 beta, CoApp 2.0.19, although you will see in a moment that that is irrelevant.

I visited the link supplied in page_url in your problem report.  This reminds me a lot of #509.  When you read that, be sure you click the links that say, "Show x previous replies."

Just visiting your web page caused VDH to list a string of items that were each 6 seconds long.  There is no point in even trying to download such items.  They are obviously not interesting.

So I went to Plan B.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/b218a2ec-fb5d-4d03-b2a2-26d77c5c71a9)

I was seeing an ever-increasing number of HLS master manifests.  This seems to be following one of the patterns I described in #509, the one where the contents of the manifest are being replaced every so often.  I'm guessing, based on the durations reported by VDH, that this replacement is happening every 6 seconds.  I noticed that even though the number of manifests seemed to be constantly increasing, the file name of the manifest seemed to be the same on each entry.  This kind of supports my hypothesis that there is only one manifest but it is being updated frequently.

I took the first HLS master manifest from the Network Monitor & ran ffprobe on it:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/15445372/ffprobe.txt)

This shows there is only 1 stream here of resolution 854x480.  Its duration is N/A.  This is a clue that this is a livestream.  Note the start value of 2086.232767.  This indicates the number of seconds that this livestream had been running at the moment I ran ffprobe.

I took this information & ran ffmpeg to download this livestream:

[Test mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/15445374/Test.mp4.Log.txt)

A few of the ffmpeg parameters I used there are worth pointing out.

-m3u8_hold_counters 25 -max_reload 25

These control the way ffmpeg detects end-of-stream.  Without these parameters, ffmpeg will read a manifest 1,000 times.  If the manifest is unchanged after those 1,000 consecutive reads, it decides the livestream has ended & it completes its recording normally.  These 2 parameters scale that back from a ridiculous 1,000 to a more reasonable 25.  However, as I'll explain in a moment, these do not come into play in this case.

-live_start_index 0

This tells ffmpeg to begin recording the livestream from the start.  Now, this depends somewhat on the server.  The server may not actually save the entire livestream from its beginning.  It may hold only an hour, 2 hours, 6 hours, some other value.  Each serving web site makes its own decision.  By specifying a value of 0 for this parameter, ffmpeg tells the server to rewind as far back as it has saved, & start feeding the livestream from there.  So it might be from the beginning or it might not.  But it's definitely a point in the past, before when I started recording with ffmpeg.  You should notice that the value of start in the ffprobe report is not the same as the start value in the ffmpeg log.  I ran ffmpeg shortly after I ran ffprobe, so the start value is larger in the ffmpeg log.  This is precisely what you would expect, since the livestream was continuing the whole time.

-t 00:05:00

This says ffmpeg will stop recording after it has received a video of duration 5 minutes.  I chose this because I didn't want to record hours upon hours of some livestream which I am utterly uninterested in.  It would be enough to get 5 minutes worth to see what was going on.  This is why my end-of-stream parameters didn't come into play.  But that's only because I limited how much I was going to record.  In the general case, you would not code -t, so the detection of end of stream will be important.

This is not how long ffmpeg is to execute.  If you look down near the bottom of the log, you'll see that this recording lasted only 4:57.  So it took ffmpeg 4:57 to record 5 minutes of this livestream.   That means the data was being transmitted at about the speed it would have if you had been simply sitting at your computer & watching the livestream in the web page.  That's a pretty slow speed, but then that's all it takes to transmit a livestream, especially since this is a rather low resolution video.

My result file looked like this:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/cb469b5c-eb2a-4606-9291-5b0e0078cdea)

And here it is playing in VLC:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/0ffda183-b27e-4534-a0f3-7f3dff546ea2)

You can see the video was just fine.  You'll have to take my word for it that there was audio, although I didn't understand the language.  I'm guessing it was Japanese.

So this shows that it is possible to record this livestream, just as I mentioned in #509.  I gave a pointer to my tutorial on ffmpeg in that other thread.  Perhaps you will have better success finding it than our fellow user in that other thread.

I admit I recorded only 5 minutes of this livestream.  Maybe I was lucky that I did not encounter the ADTS error in the 5 minutes I happened to download.  But look at these 2 discussions: #636, #588.  Both of those cases encountered what appeared to be damaged content.  But in both of those cases, waiting a few hours or maybe a day cured the problem.  There seems to be a problem trying to download something too soon after it has been made available.  The content presents itself as damaged.  But waiting a while seems to allow the web site to correct whatever the problem is & things are fine after that.

In your case here, it seems that VDH is simply incapable of dealing with the livestream.  My test shows ffmpeg has no such problem.  This is rather curious since the ffmpeg I use is the one distributed with the CoApp.  In other words, I'm using the same ffmpeg that VDH does.  So I don't know why VDH can't handle these livestreams.

The corruption in the content is a separate problem.  When that ADTS error occurs, neither VDH nor "manual" ffmpeg can get around it.  The recording fails at that point & you will have an incomplete result file.  It probably will be playable, but it won't be complete.  The only thing you can do is wait at least a few hours & try again.
